app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -51,6 +58,92 @@ angular.module('topologyApp', ['kubernetesUI', 'ui.bootstrap'])
     $interval.cancel(promise);
   });
 
+
+  var refreshProvider = function(d) {
+    $scope.refreshProvider.value = true;
+    document.getElementsByClassName('alert alert-success alert-dismissable refresh_provider')[0].style.display ='block';
```

Comment:
> Can you explain a bit about the $timeout callback part?

Sure.. if you have code like

```js
d3.select('body').on('click', function() { $scope.whatever = true; })
```

angular won't notice that $scope.whatever has changed because that code is not run as part of the digest cycle. It would notice eventually - ie. when something else happens, but that's too late. 

The way to fix that is to cause angular to notice the change.. there are more ways to do that, one being calling $scope.$digest() directly, the other being wrapping the function body in $scope.$apply, and the third one being:

```js
d3.select('body').on('click', function() { $timeout(function() {
  $scope.whatever = true;
}) });
```

Now, the `$timeout` approach is not quite the canonical approach, using `$scope.$apply` instead `$timeout` would be the "correct" way. However, both `$scope.$digest` and `$scope.$apply` have the "nice feature" that if such a call is triggered and you are already in a digest cycle, it throws. In comparison, `$timeout` just schedules a new digest cycle for as soon as possible, and runs it there, preventing that kind of problem.

(In reality, what `$timeout(callback)` does, is pretty much just `setTimeout(function() { $scope.$apply(callback); })`).